### ISSUE TITLE
server / json model does not support NaN values

### ISSUE THREAD

**Issue by gionatamettifogo** [analitico/analitico#104]
{
    "recipe_id":"rx_s24_outofstock_nb",
    "plugin":null,
    "job_id":"jb_ytWDH4RZeDEr",
    "data":[
      {
        "id":"training-samples.json",
        "created_at":"2019-04-24T06:54:11.737699+00:00",
        "filename":"training-samples.json",
        "path":"workspaces/ws_s24/models/ml_51XUatCBpDI6/data/training-samples.json",
        "hash":"e08422eb673388ee7ce2e3dea1ce09d4",
        "content_type":"application/json",
        "size":107683,
        "url":"analitico://models/ml_51XUatCBpDI6/data/training-samples.json"
      },
      {
        "id":"training.json",
        "created_at":"2019-04-24T06:54:12.835434+00:00",
        "filename":"training.json",
        "path":"workspaces/ws_s24/models/ml_51XUatCBpDI6/data/training.json",
        "hash":"35a2fa1b72f68f72b0368fc2430d2351",
        "content_type":"application/json",
        "size":7175,
        "url":"analitico://models/ml_51XUatCBpDI6/data/training.json"
      },
      {
        "id":"test.csv",
        "created_at":"2019-04-24T06:54:14.182246+00:00",
        "filename":"test.csv",
        "path":"workspaces/ws_s24/models/ml_51XUatCBpDI6/data/test.csv",
        "hash":"4bb4c823ba8fe1ba03072eb8d885dab4",
        "content_type":"text/csv",
        "size":24886,
        "url":"analitico://models/ml_51XUatCBpDI6/data/test.csv",
        "rows":200
      },
      {
        "id":"results.json",
        "created_at":"2019-04-24T06:54:15.798426+00:00",
        "filename":"results.json",
        "path":"workspaces/ws_s24/models/ml_51XUatCBpDI6/data/results.json",
        "hash":"d271ece88fe49decc47ad965eaf7ddf6",
        "content_type":"application/json",
        "size":200371,
        "url":"analitico://models/ml_51XUatCBpDI6/data/results.json"
      },
      {
        "id":"training-samples.csv",
        "created_at":"2019-04-24T06:54:16.929490+00:00",
        "filename":"training-samples.csv",
        "path":"workspaces/ws_s24/models/ml_51XUatCBpDI6/data/training-samples.csv",
        "hash":"92dd6766f58ab08e37d7c4b19933fd65",
        "content_type":"text/csv",
        "size":24370,
        "url":"analitico://models/ml_51XUatCBpDI6/data/training-samples.csv",
        "rows":200
      },
      {
        "id":"model.cbm",
        "created_at":"2019-04-24T06:54:53.777639+00:00",
        "filename":"model.cbm",
        "path":"workspaces/ws_s24/models/ml_51XUatCBpDI6/data/model.cbm",
        "hash":"6dbc5c4bf5c956ccca02c78d66e66ce7",
        "content_type":null,
        "size":45093156,
        "url":"analitico://models/ml_51XUatCBpDI6/data/model.cbm"
      }
    ],
    "training":{
      "type":"analitico/training",
      "plugins":{
        "training":"analitico.plugin.CatBoostClassifierPlugin",
        "prediction":"analitico.plugin.CatBoostClassifierPlugin"
      },
      "data":{
        "label":"dyn_purchased",
        "classes":[
          "NOT_PURCHASED",
          "PURCHASED"
        ],
        "chronological":true,
        "schema":{
          "columns":[
            {
              "name":"odt_id",
              "type":"integer"
            },
            {
              "name":"odt_ean",
              "type":"category"
            },
            {
              "name":"odt_category_id",
              "type":"category"
            },
            {
              "name":"odt_category_id.level2",
              "type":"category"
            },
            {
              "name":"odt_category_id.level3",
              "type":"category"
            },
            {
              "name":"odt_replaceable",
              "type":"category"
            },
            {
              "name":"odt_variable_weight",
              "type":"category"
            },
            {
              "name":"odt_price",
              "type":"float"
            },
            {
              "name":"odt_price_per_type",
              "type":"float"
            },
            {
              "name":"odt_surcharge_fixed",
              "type":"float"
            },
            {
              "name":"sto_name",
              "type":"category"
            },
            {
              "name":"sto_area",
              "type":"category"
            },
            {
              "name":"sto_province",
              "type":"category"
            },
            {
              "name":"sto_ref_id",
              "type":"category"
            },
            {
              "name":"odt_touched_at.year",
              "type":"category"
            },
            {
              "name":"odt_touched_at.month",
              "type":"category"
            },
            {
              "name":"odt_touched_at.day",
              "type":"category"
            },
            {
              "name":"odt_touched_at.hour",
              "type":"category"
            },
            {
              "name":"odt_touched_at.minute",
              "type":"category"
            },
            {
              "name":"odt_touched_at.dayofweek",
              "type":"category"
            },
            {
              "name":"dyn_price",
              "type":"float"
            },
            {
              "name":"dyn_price_promo",
              "type":"float"
            },
            {
              "name":"dyn_purchased",
              "type":"integer"
            }
          ]
        },
        "source_records":4670564,
        "training_records":3736452,
        "test_records":934112,
        "dropped_records":0
      },
      "parameters":{
        "test_size":0.2,
        "iterations":50,
        "learning_rate":0.75,
        "depth":8,
        "loss_function":"Logloss"
      },
      "scores":{
        "best_iteration":3,
        "best_score":{
          "training":{
            "Logloss":0.26837619434436855
          },
          "validation":{
            "Logloss":0.26041334783983155
          }
        },
        "iterations":{
          "training":{
            "Logloss":[
              0.32713041375507695,
              0.2862655052993422,
              0.27780920811841403,
              0.2725352923323516,
              0.26837619434436855,
              **NaN**
            ]
          },
          "validation":{
            "Logloss":[
              0.31260672204444423,
              0.27034562735529505,
              0.26389804385929827,
              0.26041334783983155,
              0.2604631702974502,
              NaN
            ]
          }
        },
        "features_importance":{
          "odt_ean":45.38133,
          "sto_ref_id":36.69814,
          "sto_name":8.12063,
          "odt_category_id":3.9291,
          "odt_variable_weight":3.12699,
          "odt_touched_at.hour":2.74381,
          "odt_id":0.0,
          "odt_category_id.level2":0.0,
          "odt_category_id.level3":0.0,
          "odt_replaceable":0.0,
          "odt_price":0.0,
          "odt_price_per_type":0.0,
          "odt_surcharge_fixed":0.0,
          "sto_area":0.0,
          "sto_province":0.0,
          "odt_touched_at.year":0.0,
          "odt_touched_at.month":0.0,
          "odt_touched_at.day":0.0,
          "odt_touched_at.minute":0.0,
          "odt_touched_at.dayofweek":0.0,
          "dyn_price":0.0,
          "dyn_price_promo":0.0
        },
        "log_loss":0.26041,
        "accuracy_score":0.91147,
        "precision_score_micro":0.91147,
        "precision_score_macro":0.75104,
        "precision_score_weighted":0.88807,
        "recall_score_micro":0.91147,
        "recall_score_macro":0.56643,
        "recall_score_weighted":0.91147,
        "classification_report":{
          "NOT_PURCHASED":{
            "precision":0.5829911746661003,
            "recall":0.14328299348133713,
            "f1-score":0.23003081850600074,
            "support":86214
          },
          "PURCHASED":{
            "precision":0.9190939433008041,
            "recall":0.9895789352021116,
            "f1-score":0.9530349762979883,
            "support":847898
          },
          "micro avg":{
            "precision":0.9114699308005892,
            "recall":0.9114699308005892,
            "f1-score":0.9114699308005892,
            "support":934112
          },
          "macro avg":{
            "precision":0.7510425589834522,
            "recall":0.5664309643417244,
            "f1-score":0.5915328974019946,
            "support":934112
          },
          "weighted avg":{
            "precision":0.8880732904293365,
            "recall":0.9114699308005892,
            "f1-score":0.8863052046433276,
            "support":934112
          }
        },
        "confusion_matrix":[
          [
            12353,
            73861
          ],
          [
            8836,
            839062
          ]
        ],
        "model_size":45093156
      },
      "performance":{
        "cpu_count":12,
        "training_ms":41970,
        "total_ms":100006
      },
      "algorithm":"ml/binary-classification"
    }
  }
